### PR TITLE
[feature] removing blurred frames node

### PR DESCRIPTION
**Pull request by mqcmd196**
The node for removing blurred images from image topics. When a blurred image is subscribed, nothing is published, and when a non-blurred image is subscribed, it is published as is. It detects blurred images by thresholding the value of variance of a Laplacian filtered image.

https://user-images.githubusercontent.com/27789460/186236468-98f1769d-86ba-45ed-96d9-d678795bda63.mp4

### CODE REVIEW

File: jsk_perception/sample/include/play_rosbag_fetch_go_to_kitchen.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    <remap from="~camera_info" to="$(arg QUAT_RGB_CAMERA_INFO)" />
+    <rosparam>
+      scale_width: 0.25
+      scale_height: 0.25
+    </rosparam>
+  </node>
+  <node name="resize_cloud" pkg="nodelet" type="nodelet"
+        args="load jsk_pcl/ResizePointsPublisher $(arg manager)" output="screen">
+    <remap from="~input" to="/head_camera/depth_registered/throttled/points" />
+    <remap from="~output" to="/head_camera/depth_registered/quater/throttled/points" />
```
Comment (knorth55) [inline jsk_perception/sample/include/play_rosbag_fetch_go_to_kitchen.xml:52]: points name can be changed from arguments as other topics, too?

**Comment by k-okada** [comment]
please fix tests
```
2022-08-23T19:42:43.7967374Z                                                                                 
2022-08-23T19:42:43.7967687Z Scanning dependencies of target _run_tests_jsk_perception_roslaunch-check_test_test_remove_blurred_frames.test
2022-08-23T19:42:43.7967693Z 
2022-08-23T19:42:43.7967698Z 
2022-08-23T19:42:43.7967798Z                                                                                 
2022-08-23T19:42:43.7967942Z -- run_tests.py: execute commands
2022-08-23T19:42:43.7967949Z 
2022-08-23T19:42:43.7968051Z                                                                                 
2022-08-23T19:42:43.7968369Z   /usr/bin/cmake -E make_directory /workspace/ros/ws_jsk_recognition/build/jsk_perception/test_results/jsk_perception
2022-08-23T19:42:43.7968376Z 
2022-08-23T19:42:43.7968474Z                                                                                 
2022-08-23T19:42:43.7969140Z   /opt/ros/melodic/share/roslaunch/cmake/../scripts/roslaunch-check -o "/workspace/ros/ws_jsk_recognition/build/jsk_perception/test_results/jsk_perception/roslaunch-check_test_test_remove_blurred_frames.test.xml" "/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/test_remove_blurred_frames.test" 
2022-08-23T19:42:43.7969150Z 
2022-08-23T19:42:43.7969154Z 
2022-08-23T19:42:43.7969224Z                    
2022-08-23T19:42:43.7969326Z                                                                                 
2022-08-23T19:42:43.7969419Z Traceback (most recent call last):
2022-08-23T19:42:43.7969425Z 
2022-08-23T19:42:43.7969527Z                                                                                 
2022-08-23T19:42:43.7969841Z   File "/opt/ros/melodic/share/roslaunch/cmake/../scripts/roslaunch-check", line 93, in <module>
2022-08-23T19:42:43.7969849Z 
2022-08-23T19:42:43.7969955Z                                                                                 
2022-08-23T19:42:43.7970202Z     error_msg = check_roslaunch_file(roslaunch_path, use_test_depends=options.test_depends, ignore_unset_args=options.ignore_unset_args)
2022-08-23T19:42:43.7970210Z 
2022-08-23T19:42:43.7970314Z                                                                                 
2022-08-23T19:42:43.7970614Z   File "/opt/ros/melodic/share/roslaunch/cmake/../scripts/roslaunch-check", line 47, in check_roslaunch_file
2022-08-23T19:42:43.7970621Z 
2022-08-23T19:42:43.7970626Z 
2022-08-23T19:42:43.7970726Z                                                                                 
2022-08-23T19:42:43.7970971Z     error_msg = roslaunch.rlutil.check_roslaunch(roslaunch_file, use_test_depends=use_test_depends, ignore_unset_args=ignore_unset_args)
2022-08-23T19:42:43.7970977Z 
2022-08-23T19:42:43.7971081Z                                                                                 
2022-08-23T19:42:43.7971374Z   File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/rlutil.py", line 202, in check_roslaunch
2022-08-23T19:42:43.7971381Z 
2022-08-23T19:42:43.7971472Z                                                                                 
2022-08-23T19:42:43.7971732Z     base_pkg, file_deps, missing = roslaunch.depends.roslaunch_deps([f], use_test_depends=use_test_depends, ignore_unset_args=ignore_unset_args)
2022-08-23T19:42:43.7971739Z 
2022-08-23T19:42:43.7971842Z                                                                                 
2022-08-23T19:42:43.7972134Z   File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/depends.py", line 341, in roslaunch_deps
2022-08-23T19:42:43.7972141Z 
2022-08-23T19:42:43.7972146Z 
2022-08-23T19:42:43.7972251Z                                                                                 
2022-08-23T19:42:43.7972376Z     raise RoslaunchDepsException(str(e))
2022-08-23T19:42:43.7972383Z 
2022-08-23T19:42:43.7972487Z                                                                                 
2022-08-23T19:42:43.7972678Z roslaunch.depends.RoslaunchDepsException: No value for arg [rosbag]
2022-08-23T19:42:43.7972728Z 
2022-08-23T19:42:43.7972733Z 
2022-08-23T19:42:43.7972836Z                                                                                 
2022-08-23T19:42:43.7973308Z Cannot find results, writing failure results to '/workspace/ros/ws_jsk_recognition/build/jsk_perception/test_results/jsk_perception/MISSING-roslaunch-check_test_test_remove_blurred_frames.test.xml'
2022-08-23T19:42:43.7973316Z 
2022-08-23T19:42:43.7973321Z 
2022-08-23T19:42:43.7973428Z                                                                                 
2022-08-23T19:42:43.7973726Z [31mMissing package dependencies: jsk_perception/package.xml: jsk_tools (notify upstream maintainer)[0m
2022-08-23T19:42:43.7973733Z 
2022-08-23T19:42:43.7973741Z 
2022-08-23T19:42:43.7973834Z                                                                                 
2022-08-23T19:42:43.7974119Z [31mMissing package dependencies: jsk_perception/package.xml: jsk_tools (notify upstream maintainer)[0m
2022-08-23T19:42:43.7974125Z 
2022-08-23T19:42:43.7974232Z                                                              
2022-08-23T19:42:43.7974450Z checking /workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/test_remove_blurred_frames.test
2022-08-23T19:42:43.7974457Z 
2022-08-23T19:42:43.7974462Z 
2022-08-23T19:42:43.7974567Z                                                                                 
2022-08-23T19:42:43.7974986Z -- run_tests.py: verify result "/workspace/ros/ws_jsk_recognition/build/jsk_perception/test_results/jsk_perception/roslaunch-check_test_test_remove_blurred_frames.test.xml"
2022-08-23T19:42:43.7974993Z 
2022-08-23T19:42:43.7974998Z 
2022-08-23T19:42:43.7975152Z                                                                                 
2022-08-23T19:42:43.7975431Z Built target _run_tests_jsk_perception_roslaunch-check_test_test_remove_blurred_frames.test
2022-08-23T19:42:43.7975438Z 
2022-08-23T19:42:43.7975442Z 
2022-08-23T19:42:43.7975545Z                                                                                 
2022-08-23T19:42:43.7975848Z Scanning dependencies of target _run_tests_jsk_perception_roslaunch-check_test_consensus_tracking.test
2022-08-23T19:42:43.7975855Z 
2022-08-23T19:42:43.7975860Z 
2022-08-23T19:42:43.7975970Z                                                                                 
```

File: jsk_perception/src/remove_blurred_frames.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        cv::Mat image, gray, laplacianImage;
+        cv::Scalar mean, stddev;
+        double var;
+        vital_checker_ -> poke();
+        boost::mutex::scoped_lock lock(mutex_);
+        image = cv_bridge::toCvShare(image_msg, image_msg->encoding) -> image;
+        cv::cvtColor(image, gray, cv::COLOR_RGB2GRAY);
+        cv::Laplacian(gray, laplacianImage, CV_64F);
+        cv::meanStdDev(laplacianImage, mean, stddev, cv::Mat());
+        var = stddev.val[0] * stddev.val[0];
```
Comment (k-okada) [inline jsk_perception/src/remove_blurred_frames.cpp:71]: we might want to know the current variable and threshould, so how about something like
```
ROS_DEBUG("%s : Check variance of a Laplacian: %f > %f",__FUNCTION__, var, min_laplacian_var_);
```

File: jsk_perception/src/remove_blurred_frames.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+#include "jsk_perception/remove_blurred_frames.h"
+
+namespace jsk_perception{
+    void RemoveBlurredFrames::onInit(){
+        DiagnosticNodelet::onInit();
+        srv_ = std::make_shared <dynamic_reconfigure::Server<Config> > (*pnh_);
+        dynamic_reconfigure::Server<Config>::CallbackType f =
+            boost::bind(&RemoveBlurredFrames::configCallback, this, _1, _2);
+        srv_->setCallback (f);
+        pub_ = advertise<sensor_msgs::Image>(*pnh_, "output", 1);
```
Comment (iory) [inline jsk_perception/src/remove_blurred_frames.cpp:49]: How about publish mask image like in the next repository?
https://github.com/WillBrennan/BlurDetection2

**Review by iory — CHANGES_REQUESTED** [review]

File: jsk_perception/sample/include/play_rosbag_fetch_go_to_kitchen.xml (hunk 1 above)
Comment (mqcmd196) [reply jsk_perception/sample/include/play_rosbag_fetch_go_to_kitchen.xml]: Yes, correct. I updated.

File: jsk_perception/src/remove_blurred_frames.cpp (hunk 2 above)
Comment (mqcmd196) [reply jsk_perception/src/remove_blurred_frames.cpp:73]: I agreed, updated

File: jsk_perception/src/remove_blurred_frames.cpp (hunk 3 above)
Comment (mqcmd196) [reply jsk_perception/src/remove_blurred_frames.cpp:49]: I fixed to publish mask image

Comment (iory) [reply jsk_perception/src/remove_blurred_frames.cpp:49]: Great!

File: jsk_perception/src/remove_blurred_frames.cpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        image = cv_bridge::toCvShare(image_msg, image_msg->encoding) -> image;
+        cv::cvtColor(image, gray, cv::COLOR_RGB2GRAY);
+        cv::Laplacian(gray, laplacian_image, CV_64F);
+        cv::meanStdDev(laplacian_image, mean, stddev, cv::Mat());
+        var = stddev.val[0] * stddev.val[0];
+        NODELET_DEBUG("%s : Checking variance of the Laplacian: actual %f, threshold %f\n", __func__, var, min_laplacian_var_);
+        laplacian_image.convertTo(masked_image, CV_8UC1, 255.0);
+        mask_img_msg = cv_bridge::CvImage(image_msg->header, sensor_msgs::image_encodings::MONO8, masked_image).toImageMsg();
+        pub_masked_.publish(mask_img_msg);
+        if(var > min_laplacian_var_){
```
Comment (iory) [inline jsk_perception/src/remove_blurred_frames.cpp:78]: It may be that the user wants to use this value, so it is better to publish score (`var`).

**Review by iory — CHANGES_REQUESTED** [review]

File: jsk_perception/src/remove_blurred_frames.cpp (hunk 4 above)
Comment (mqcmd196) [reply jsk_perception/src/remove_blurred_frames.cpp:82]: updated

File: jsk_perception/src/remove_blurred_frames.cpp (hunk 3 above)
Comment (iory) [reply jsk_perception/src/remove_blurred_frames.cpp:49]: I created the PR to fix the mask image.
https://github.com/mqcmd196/jsk_recognition/pull/2
Please confirm it.

Comment (iory) [reply jsk_perception/src/remove_blurred_frames.cpp:49]: Screenshot 2022-08-24 18:48:07
We can get an image like this.

**Comment by mqcmd196** [comment]
Test failes in indigo
```traivs_fold:start:catkin_test_results
traivs_time:start:170ecbbcf5d25045>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+ catkin_test_results --verbose --all build
Skipping "catkin_tools_prebuild/package.xml": 'tests'
Skipping "jsk_perception/test_results/jsk_perception/rosunit-test_draw_rects_py.xml": no element found: line 2, column 0
Skipping "jsk_perception/test_results/jsk_perception/rosunit-test_face_pose_estimation.xml": no element found: line 2, column 0
Skipping "jsk_perception/test_results/jsk_perception/rosunit-test_people_pose_estimation_2d.xml": no element found: line 2, column 0
Skipping "jsk_perception/test_results/jsk_perception/rosunit-test_human_mesh_recovery.xml": no element found: line 2, column 0
Full test results for 'jsk_perception/test_results/jsk_perception/rosunit-test_remove_blurred_frames.xml'
-------------------------------------------------
<?xml version="1.0" encoding="utf-8"?>
<testsuite errors="0" failures="1" name="unittest.suite.TestSuite" tests="1" time="10.012">
  <testcase classname="__main__.TestTopicPublished" name="test_published" time="10.0122">
    <failure type="AssertionError">Timed out (10s) of /clock publication.
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/opt/ros/indigo/share/jsk_tools/src/test_topic_published.py", line 70, in test_published
    self.fail('Timed out (10s) of /clock publication.')
  File "/usr/lib/python2.7/unittest/case.py", line 410, in fail
    raise self.failureException(msg)
    </failure>
  </testcase>
  <system-out><![CDATA[]]></system-out>
  <system-err><![CDATA[[WARN] [WallTime: 1661486870.325766] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486871.326468] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486872.328080] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486873.329841] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486874.330453] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486875.331913] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486876.332650] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486877.333439] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486878.334217] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486879.335711] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
[WARN] [WallTime: 1661486880.336455] [0.000000] /use_sim_time is specified and rostime is 0, /clock is published?
]]></system-err>
</testsuite>

-------------------------------------------------
Skipping "jsk_pcl_ros_utils/test_results/jsk_pcl_ros_utils/rosunit-test_pointcloud_to_point_indices.xml": no element found: line 2, column 0
```
Should I skip the test in indigo... ???

**Comment by knorth55** [comment]
this is weird. can you check this test error in local docker image?

**Comment by iory** [comment]
It seems that `gdown` failed to download.
```
2022-08-26T12:02:21.6380911Z Traceback (most recent call last):
2022-08-26T12:02:21.6381040Z   File "/usr/local/bin/gdown", line 5, in <module>
2022-08-26T12:02:21.6381185Z     from pkg_resources import load_entry_point
2022-08-26T12:02:21.6381497Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2749, in <module>
2022-08-26T12:02:21.6381640Z     working_set = WorkingSet._build_master()
2022-08-26T12:02:21.6381965Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 444, in _build_master
2022-08-26T12:02:21.6382074Z     ws.require(__requires__)
2022-08-26T12:02:21.6382380Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 725, in require
2022-08-26T12:02:21.6382557Z     needed = self.resolve(parse_requirements(requirements))
2022-08-26T12:02:21.6382856Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 633, in resolve
2022-08-26T12:02:21.6383118Z     requirements.extend(dist.requires(req.extras)[::-1])
2022-08-26T12:02:21.6383486Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2291, in requires
2022-08-26T12:02:21.6383587Z     dm = self._dep_map
2022-08-26T12:02:21.6383901Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2484, in _dep_map
2022-08-26T12:02:21.6384052Z     self.__dep_map = self._compute_dependencies()
2022-08-26T12:02:21.6384406Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2517, in _compute_dependencies
2022-08-26T12:02:21.6384533Z     common = frozenset(reqs_for_extra(None))
2022-08-26T12:02:21.6384861Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2514, in reqs_for_extra
2022-08-26T12:02:21.6385079Z     if req.marker_fn(override={'extra':extra}):
2022-08-26T12:02:21.6385410Z   File "/usr/lib/python2.7/dist-packages/_markerlib/markers.py", line 113, in marker_fn
2022-08-26T12:02:21.6385560Z     return eval(compiled_marker, environment)
2022-08-26T12:02:21.6385706Z   File "<environment marker>", line 1, in <module>
2022-08-26T12:02:21.6385933Z NameError: name 'platform_system' is not defined
2022-08-26T12:02:21.6386058Z Traceback (most recent call last):
2022-08-26T12:02:21.6386188Z   File "/usr/local/bin/gdown", line 5, in <module>
2022-08-26T12:02:21.6386333Z     from pkg_resources import load_entry_point
2022-08-26T12:02:21.6386648Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2749, in <module>
2022-08-26T12:02:21.6386793Z     working_set = WorkingSet._build_master()
2022-08-26T12:02:21.6387117Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 444, in _build_master
2022-08-26T12:02:21.6387226Z     ws.require(__requires__)
2022-08-26T12:02:21.6387535Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 725, in require
2022-08-26T12:02:21.6387698Z     needed = self.resolve(parse_requirements(requirements))
2022-08-26T12:02:21.6388006Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 633, in resolve
2022-08-26T12:02:21.6388268Z     requirements.extend(dist.requires(req.extras)[::-1])
2022-08-26T12:02:21.6388581Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2291, in requires
2022-08-26T12:02:21.6388676Z     dm = self._dep_map
2022-08-26T12:02:21.6388984Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2484, in _dep_map
2022-08-26T12:02:21.6389135Z     self.__dep_map = self._compute_dependencies()
2022-08-26T12:02:21.6389549Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2517, in _compute_dependencies
2022-08-26T12:02:21.6389675Z     common = frozenset(reqs_for_extra(None))
2022-08-26T12:02:21.6390001Z   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2514, in reqs_for_extra
2022-08-26T12:02:21.6390215Z     if req.marker_fn(override={'extra':extra}):
2022-08-26T12:02:21.6390538Z   File "/usr/lib/python2.7/dist-packages/_markerlib/markers.py", line 113, in marker_fn
2022-08-26T12:02:21.6390683Z     return eval(compiled_marker, environment)
2022-08-26T12:02:21.6390826Z   File "<environment marker>", line 1, in <module>
2022-08-26T12:02:21.6391054Z NameError: name 'platform_system' is not defined
```

**Comment by mqcmd196** [comment]
It seems to fail downloading in indigo 
```
[FATAL] [1661927053.441860318]: Error opening file: /workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/sample/data/fetch_go_to_kitchen.bag
```

**Comment by mqcmd196** [comment]
All test OK. Thank you for rerunning

**Review by knorth55 — APPROVED** [review]

**Review by iory — APPROVED** [review]

File: jsk_perception/include/jsk_perception/remove_blurred_frames.h [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/o2r other materials provided
```
Comment (k-okada) [inline jsk_perception/include/jsk_perception/remove_blurred_frames.h:16]: please fix type 'o2r', see https://github.com/jsk-ros-pkg/jsk_recognition/pull/2727

File: jsk_perception/src/remove_blurred_frames.cpp [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+    void RemoveBlurredFrames::work(const sensor_msgs::Image::ConstPtr& image_msg){
+        cv::Mat image, gray, laplacian_image;
+        sensor_msgs::Image::Ptr mask_img_msg;
+        cv::Scalar mean, stddev;
+        std_msgs::Float64 var_msg;
+        double var;
+        vital_checker_ -> poke();
+        boost::mutex::scoped_lock lock(mutex_);
+        image = cv_bridge::toCvShare(image_msg, image_msg->encoding) -> image;
```
Comment (nakane11) [inline jsk_perception/src/remove_blurred_frames.cpp:71]: Could you specify rgb encoding? R and B seem to be incompatible.
> RGB[A] to Gray:Y←0.299⋅R+0.587⋅G+0.114⋅B

https://docs.opencv.org/4.1.0/de/d25/imgproc_color_conversions.html
https://github.com/opencv/opencv/blob/347246901eccabe503985a64f16813ca859af25a/modules/imgproc/src/color.cpp#L211-L214

File: jsk_perception/include/jsk_perception/remove_blurred_frames.h (hunk 5 above)
Comment (mqcmd196) [reply jsk_perception/include/jsk_perception/remove_blurred_frames.h]: fixed 311a6618e2f531a992f6b7ce189d5f1527128eb1

File: jsk_perception/src/remove_blurred_frames.cpp (hunk 6 above)
Comment (mqcmd196) [reply jsk_perception/src/remove_blurred_frames.cpp:71]: Do you mean as both `COLOR_BGR2GRAY` and `COLOR_RGB2GRAY` use same code, we shouldn't use `cv::cvtColor`

Comment (mqcmd196) [reply jsk_perception/src/remove_blurred_frames.cpp:71]: And, is it the bug of opencv?

Comment (mqcmd196) [reply jsk_perception/src/remove_blurred_frames.cpp:71]: https://github.com/opencv/opencv/blob/17234f82d025e3bbfbf611089637e5aa2038e7b8/modules/imgproc/src/color.hpp#L54 ??

Comment (nakane11) [reply jsk_perception/src/remove_blurred_frames.cpp:71]: To convert RGB to GRAY in the next line, the encoding have to be RGB, not as input is.

Comment (nakane11) [reply jsk_perception/src/remove_blurred_frames.cpp:71]: divide the next line like RGB2GRAY and BGR2GRAY, or specify RGB

Comment (mqcmd196) [reply jsk_perception/src/remove_blurred_frames.cpp:71]: I see, the line `cv::cvtColor(image, gray, cv::COLOR_RGB2GRAY);` . Sorry for my overlooking.

**Review by nakane11 — COMMENTED** [review]
LGTM!
bgr8, bgr16, and bgra8 images respectively returned the same variance as rgb images .

**Pull request merged**